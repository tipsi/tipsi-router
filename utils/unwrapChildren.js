import React from 'react'

export default function unwrapChildren(routes, acc = []) {
  return routes.reduce((memo, { Route, path, children }, index) => {
    if (!children || Array.isArray(children) && !children.length) {
      return memo.concat(
        <Route path={path} key={path} />
      )
    }

    return memo.concat(
      <Route path={path} key={path}>
        {unwrapChildren(children)}
      </Route>
    )
  }, acc)
}
