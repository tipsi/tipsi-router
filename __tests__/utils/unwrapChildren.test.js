import React from 'react'
import { unwrapChildren } from '../../utils'

describe('unwrapChildren', () => {
  it('should unwrap children recursively', () => {
    const Route = () => <div />

    const routes = [{
      Route: children => children,
      path: '/',
      children: [
        { Route, path: 'login' },
        {
          Route: children => children,
          path: 'about',
          children: [],
        },
        {
          Route: children => children,
          path: 'stores',
          children: [{ Route, path: ':storeId' }],
        },
      ],
    }]

    expect(unwrapChildren(routes)).toMatchSnapshot()
  })
})
