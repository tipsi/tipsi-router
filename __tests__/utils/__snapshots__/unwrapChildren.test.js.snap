// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unwrapChildren should unwrap children recursively 1`] = `
Array [
  <Route
    path="/"
  >
    <Route
      path="login"
    />
    <Route
      path="about"
    />
    <Route
      path="stores"
    >
      <Route
        path=":storeId"
      />
    </Route>
  </Route>,
]
`;
